### ISSUE TITLE
Renaming "Security" admin area to "People"

### ISSUE THREAD

**Issue by clarkepaul** [silverstripe/silverstripe-framework#8662]
## Affected Version

All versions

## Description

This might be a slightly contentious topic so we would welcome feedback or input from others on this topic.

While doing research (for how people use/setup site permissions and user permissions) we found that "Security" might not be the best wording to indicate a member management area. Although you can administer groups of people by assigning roles and permissions the main thing you are managing is people and their roles within the CMS. The "Security" section isn't an area to administer general website security settings like the wording might suggest, you do site wide security settings in "Settings/Access" area of the CMS.

Also the member database isn't just used to store Admin users but could also extend to other list of people (eg. newsletter signups, contacts, staff members). For these larger member lists which include more people than just CMS administrators you generally don't apply permissions/roles to them, therefor the term "Security" doesn't make as much sense.

Our recommendation is to rename the "Security" section to "People", we did also consider "Members" but didn't like the sound of that. People seems fairly inclusive no matter how people might use the section and should clarify that it isn't a general website security setting section.

cc @chillu

**Comment by newleeland** [comment]
**MFA related benefits of changing name**
The "Security" terminology also confused admins trying to enable MFA on their CMS setup, during MFa testing. Some admins went to "Security" rather than "Settings". When questioned they originally intended to go to "Settings" but saw security and decided that sounded more like an area for MFA.

**Comment by kinglozzer** [comment]
I prefer “People“ to “Security”, but I’m still not convinced it describes that admin section very well - if we’re keeping groups and permissions stuff there too, I think we need to communicate that somehow. My suggestions: “Users & Groups” or “User Management”.

**Comment by phptek** [comment]
What about SilverStripe web applications that manage hardware and software "users"? For example an LDAP management application for managing routers and other network hardware, or a tool that manages IoT devices? In that case, the word "Accounts" makes more sense, and IMO covers off humans _and_ artifical systems.

**Comment by maxime-rainville** [comment]
Totally agree with renaming "Security" to "People". Generally, I'm not a fan of calling any specific section of a system "Security" as it implies security is some sort of addon that can be neatly contained in one specific area.

@kinglozzer  I'm not sure it's worthwhile having a more complex expression like "User Management" or "User and Groups". The main object we're tracking here is `Member`. `Groups` makes no sense without `Member`

@chillu Agree that this change might be introduce confusion between Member and People. If we come to the conclusion that People/Person is better, we could just update the translations for `Member` while keeping the internal name the same.

@phptek 

> What about SilverStripe web applications that manage hardware and software "users"?
For these less common use cases, we could encourage developers to set their own label that might be more descriptive.

**Comment by chillu** [comment]
@clarkepaul I think we're waiting for your response on how to handle the other cases raised here :)

**Comment by clarkepaul** [comment]
Cool, based on the discussion here I think that going ahead with the wording "People" makes the most sense. I think there might be cases where "Member" might still be the most appropriate language to use so it isn't a matter of find and replace. I've scanned most of the CMS and I can't find a place where we would have a problem yet. 

For now I would leave all of the references to "Users" throughout the CMS as it still makes sense, although I'm not a fan of that word as it also has negative connotations, but sometimes it makes writing a lot easier.
![image](https://user-images.githubusercontent.com/555033/52239475-771d7180-2933-11e9-8453-f1e4ed67a47e.png)

"People" would work in this scenario:
![image](https://user-images.githubusercontent.com/555033/52238662-7257be00-2931-11e9-979e-8c0a247a134f.png)

"Add new" would work in this scenario:
![image](https://user-images.githubusercontent.com/555033/52238777-ba76e080-2931-11e9-8728-197a9a79cff0.png)

"Member" still makes sense in the context of belonging to a group (member of a group):
![image](https://user-images.githubusercontent.com/555033/52238877-fc078b80-2931-11e9-9f1d-7693526e8225.png)

Search for "People" would work here:
![image](https://user-images.githubusercontent.com/555033/52238950-29543980-2932-11e9-9263-078b0c080488.png)

**Comment by clarkepaul** [comment]
@chillu would this be a breaking change or could we get this type of thing included in a minor release? Someone mentioned we could perhaps use an alias in code if it was an issue.

**Comment by chillu** [comment]
My feeling is that would be a breaking change, but we could document how to change it back in the upgrading guide if your authors feel strongly about it - let's get this into 4.x

**Comment by ScopeyNZ** [comment]
> let's get this into 4.x

Tagged as minor as such. Relabelling isn't a major in my eyes. We can't consider the names of our pages/nav as part of our API 😆 .

Changing `Member` (the class) is a different story of course...

**Comment by chillu** [comment]
Hrm so we're moving from "security, users, members" wording to "people, users, members" wording, with "add member" replaced by "add new" because "add people" would be awkward? That doesn't seem like it's improving consistency a whole lot, just just replacing three variations with three slightly different variations.

**Comment by clarkepaul** [comment]
Thats the pattern which is used in other areas. "Add new" relates to the section you are in. A lot of UIs go even simpler with just a plus icon or similar. The main issue was the word "Security" not the rest of them. As I said above member still makes sense as being a member of a group.
![image](https://user-images.githubusercontent.com/555033/53618667-36bec400-3c51-11e9-86c0-0b6a58e9c447.png)

**Comment by michalkleiner** [comment]
I agree with using generic Add new, but I don't think it should be labeled "People".
My (personal) preference, seeing the Member object may go beyond people, pets or machines to hold information for, also not necessarily implying these entities will use it to log into the CMS.

I'd answer these questions first before trying to find a single name covering everything — What does the section cover? Would it be better split into multiple, more focused sections?

As I see it, it serves for CMS accounts management for, security groups and roles management, permissions configuration, so that all is related to _access_ and _security_. Members (as in the data holders) can be tied to several different auth mechanisms (RealMe, OAuth, LDAP, JWT) etc., may be created automatically or from the front-end. Might be service accounts for scheduled tasks.

Also, how do other CMS systems name such a section? It's not always the best to go with a unique approach if there's something generally accepted.

**Comment by clarkepaul** [comment]
"Users" seems to be the most common term used in other CMSs, "People" is also used (e.g. Wordpress), "Security" I haven't seen as much but it does make sense in a lot of ways. I think "Security" has been the old fashioned naming convention? I personally don't like to just use the word "users" without providing extra context like "CMS users" as it can often be used as a negative term, but I do think its more appropriate than security.

**Comment by micschk** [comment]
**TLDR;** I initially wanted to make the point that **Acounts** may be a more appropriate term but then considered a current situation where one *Account* may have multiple *Users*, so to me actually **Users makes more sense after all**.

---

A year and a bit late but for me the term 'users' means the *people* using a website or application. These people may or may not have an account/login credentials. Also one person may have multiple different accounts, eg one for daily use and one for administrative tasks.

Since we're not actually managing *people* but their *login credentials and privileges*, the term **Accounts** (or Accounts & Permissions) would seem more appropriate to me. This would also fit situations where you're managing API credentials or machine/service accounts etc as well (basically what @phptek said 👍).

One slightly more complex situation which we have at hand now is a system in which multiple Users may act on behalf of one or multiple Accounts (so here an *Account* does not map to a person, but is more of a business entity which multiple *Users* may share access to).

Looking at Google Cloud, they use 'AIM' as nav/menu term (Identity Access Management).  
And as table headers:
- Type ('Service Account' for machine accounts & 'User' for humans)
- **Member** (for the username, so 'Username' would make more sense imo)
- Name
- Role
- (granted permissions)

**Comment by clarkepaul** [comment]
Some good points, "Users" is certainly an improvement to what we've currently got. For me it's still a close one between "Users" and "People". Even with other account info which might not even be people based. "Users" covers the different scenarios better, and "People" looks and reads better to the bulk of use cases IMHO.

Unfortunately I think "AIM" might be a stretch too far for our audience, but some really good terms for consideration in the long form of the name.

Not a fan of "Accounts" personally although I can see a lot of reasons why it might be used technically. For me it has too many ties to financial accounts, or "account management" of businesses and not individuals. Not sure if this would conflict with individual site data and terminologies used. Probably not the most friendliest term for CMS users either. I can see how "accounts" would related to a lot of websites front-end though... Sign-up for a new account etc.

**Comment by madmatt** [comment]
Removed `effort/easy` label given @clarkepaul added the `effort/medium` label :-)

**Comment by GuySartorelli** [comment]
I would personally prefer erring away from "people" or "users". I am aware of a few sites that have a separate section for front-end users which could easily get confused with "people" or "users", and another that holds data about individuals which again would be confused with "people".

**Comment by micschk** [comment]
How about just moving all of this into a config block so devs can pick the
most appropriate wording for their specific situation?

**Comment by GuySartorelli** [comment]
> How about just moving all of this into a config block so devs can pick the most appropriate wording for their specific situation?

It is tech ideally all configurable now via i18n I believe - so the question is ultimately what the default value should be.
